src/hex.rs

```diff
@@ -64,9 +64,9 @@ impl<T: Hash> FromHex for T {
 
 /// Iterator over a hex-encoded string slice which decodes hex and yields bytes.
 pub struct HexIterator<'a> {
-    /// The `Chars` iterator whose first two characters will be decoded to yield
+    /// The `Bytes` iterator whose next two bytes will be decoded to yield
     /// the next byte.
-    chars: Chars<'a>,
+    iter: str::Bytes<'a>,
```

Comment:
I think the motivation was something with multi-byte characters that mess up the `.len()` check for oddness in the beginning.

Reply:
Ah, right, we were using `.len()` insted of `.chars().count()` (it's kinda unintuitive that they aren't the same).

Reply:
It's because `.len()` is O(1) while `chars().count()` is O(n).

I am pretty surprised that the byte iterator is slower than the chars iterator, especially by such a large factor. I suspect this is a bug in stdlib.